raiden/network/rpc/client.py

```diff
@@ -70,6 +70,24 @@ def check_transaction_threw(client, transaction_hash):
     return int(transaction['gas'], 0) == int(receipt['gasUsed'], 0)
 
 
+def retry_if_connnection_error(tries, reconnect_wait_secs, method, *args, **kwargs):
+    tries = 5
+    reconnect_wait_secs = 4
+    while tries != 0:
+        try:
+            result = method(*args, **kwargs)
+            break
+        except requests.exceptions.ConnectionError:
+            log.info(
+                'Timeout in eth client connection. Is client offline? Trying'
+                ' again in {} secs. {} tries remaining'.format(reconnect_wait_secs, tries),
+            )
+            gevent.sleep(reconnect_wait_secs)
```

Comment:
Should we the reconnect trials non-linear? Often something exponential with a limit is used.